### DISCUSSION TITLE
Azure SQL support for persistence in Cadence

### DISCUSSION THREAD

**Discussion by sairamkunche** [uber/cadence#5603] [General]
"Hello Cadence community! I'm exploring the possibility of using Azure SQL for persistence in my Cadence deployment instead of a Cassandra container. I would appreciate any insights or guidance on the following:

**Integration Steps:**

Has anyone successfully configured Cadence to use Azure SQL for persistence? If so, could you share the key steps involved in the integration process?

**Configuration Settings:**

Are there specific configuration settings or considerations I should be aware of when setting up Cadence to use Azure SQL as the persistence layer?

Could you point me to any relevant documentation, guides, or community-contributed resources that specifically cover integrating Cadence with Azure SQL for persistence?
Thank you in advance for any assistance or insights you can provide!"

**Comment by Shaddoll** [comment]
Here is a doc you can follow: https://github.com/uber/cadence/blob/master/docs/persistence.md

In summary, you need to:
1. write a new Azure SQL [plugin](https://github.com/uber/cadence/tree/master/common/persistence/sql/sqlplugin) for Cadence;
2. run the existing [persistence](https://github.com/uber/cadence/tree/master/common/persistence/persistence-tests) and [integration](https://github.com/uber/cadence/tree/master/host) tests with new database;
3. [Register](https://github.com/uber/cadence/blob/master/common/persistence/sql/sqlplugin/mysql/plugin.go#L67) the new plugin in your binary and set up connection [config](https://github.com/uber/cadence/blob/master/common/config/config.go#L288)
4. And then you can write your docker-compose with Azure SQL dependency
